Solutions/Marain.UserNotifications.Abstractions/Marain/NotificationTemplates/CommunicationTemplates/EmailTemplate.cs

```diff
@@ -0,0 +1,43 @@
+﻿// <copyright file="EmailTemplate.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.NotificationTemplates.CommunicationTemplates
+{
+    /// <summary>
+    /// Email Object.
+    /// </summary>
+    public class EmailTemplate : ICommunicationTemplate
+    {
+        /// <summary>
+        /// The content type that will be used when serializing/deserializing.
+        /// </summary>
+        public const string RegisteredContentType = "application/vnd.marain.usernotifications.notificationtemplate.emailtemplate.v1";
+
+        /// <inheritdoc/>
+        public string? NotificationType { get; set; }
+
+        /// <inheritdoc/>
+        public string? ETag { get; set; }
+
+        /// <summary>
+        /// Gets or sets the body of the email.
+        /// </summary>
+        public string? Body { get; set; }
+
+        /// <summary>
+        /// Gets or sets the subject of the email.
+        /// </summary>
+        public string? Subject { get; set; }
```

Comment:
It feels like "no subject" is better modelled as String.Empty and you should be required to state that explicitly, so this should not be Nullable.

Reply:
We did this as if we do not keep these nullable we would have to populate them with the constructor.

https://gifyu.com/image/UQiL

Reply:
That feels like a feature, not a bug!

Reply:
OK, that's not a good reason for making it nullable.

As it happens, last year I wrote a blog on the very issue you've run into here: https://endjin.com/blog/2020/09/dotnet-csharp-8-nullable-references-serialization

I would recommend that you read the whole thing because it outlines the various possible responses to this issues, and their pros and cons. (And it includes an explanation of why making a property nullable purely to get rid of the warning is a bad idea.)

But the key points are: 1) what's wrong with constructor initialization? In most cases, that's actually preferable. As a general rule you should use it unless for some reason you can't. And 2) if for some reason you simply can't use the "correct by construction" approach, and if you feel it really isn't worth the effort going down the  "Nullable backing field, non-nullable property" path, then you're usually actually better off just admitting that you've given up, and using a suitable `#nullable` directive for the properties you've decided aren't really going to participate fully in nullability. You can basically opt the property out—disable nullable annotations for just that line of code, and C# will now treat just that one line of code as "null oblivious".

But constructor initialization should definitely be your first choice. (And I'd go further: does this actually need to be a mutable class? You tend to get the most benefit from nullable references if you make plain-old-data-like classes such as these immutable.)

Reply:
Thank you for the blog. We have read it and makes more sense now :)

Reply:
Thanks for that Ian, we were a little unsure on the pattern of using a constructor implementation to create objects, but your post cleared up quite a few things.

Reply:
Hi all,
As Steve had discussed with Matthew and Jon, we shall create a issue in the github and we shall look at this once the 1st slice has been integrated.

Reply:
Can you create the issue and reference the issue number here please? Once that's done we'll resolve this conversation.

Reply:
Hi all, as promised we have created the [issue](https://github.com/marain-dotnet/Marain.UserNotifications/issues/123).